app/src/main/java/com/onboarding/calculator/mvp/model/CalculatorModel.java

```diff
@@ -132,22 +138,40 @@ public Double getResult() {
                     break;
                 }
             }
+        } else {
+            error = Error.ERROR_INCOMPLETE_OPERATION;
+            result = null;
         }
         reset();
+        if (result != null) {
+            firstOperand = result.toString();
+        }
         return result;
     }
 
-    @Override
-    public String getOperator() {
-        return operator;
+    public Error getError() {
+        Error aux = error;
+        error = Error.NONE;
+        return aux;
```

Comment:
split the aux value into the return sentence

Reply:
Solved!